cmd/jujud/updateseries/updateseries.go

```diff
@@ -118,38 +139,64 @@ func (c *UpdateSeriesCommand) Run(ctx *cmd.Context) error {
 		return err
 	}
 
+	jujuVersion, err := GetJujuVersion(c.machineAgent, c.dataDir)
+	if err != nil {
+		return err
+	}
+
 	switch {
 	case toInitSys == service.InitSystemUpstart:
 		return errors.NewNotSupported(nil, "downgrade to series using upstart not supported")
 	case toInitSys == fromInitSys && toInitSys == service.InitSystemSystemd:
-		if err = c.copyTools(); err != nil {
+		if err = service.CopyAgentBinary(c.machineAgent, c.unitAgents, c.dataDir, c.toSeries, c.fromSeries, jujuVersion); err != nil {
 			return err
 		} else {
 			ctx.Infof("successfully copied tools and relinked agent tools")
 		}
 		if !c.startAgents {
 			break
 		}
-		if err = c.startAllAgents(ctx); err != nil {
+		if errMachineName, errUnitNames, err = service.StartAllAgents(c.machineAgent, c.unitAgents, c.dataDir, c.toSeries); err != nil {
```

Comment:
This should use a := to define these variables here - this limits the scope of them to the if statement. Then you don't need to declare them up at the top of the method. Also the names variables shouldn't be `errMachineName` and `errUnitNames`, since they're values that are set when there isn't an error.